src/thread.rs

```diff
@@ -332,7 +332,6 @@ where
 
         if let ThreadStatus::Resumable = self.thread.status() {
             // Ignore value returned via yield()
-            cx.waker().wake_by_ref();
```

Comment:
We need to keep this `wake_by_ref` mostly to be able to schedule "manual" futures represented as coroutines executed in async mode.
See: https://github.com/khvzak/mlua/blob/master/tests/async.rs#L89
This test would fail without this `wake_by_ref`.